servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SourceAdapters.java

```diff
@@ -133,39 +135,75 @@ private static <T> SingleSource<T> uncheckedCast(final Single<T> single) {
         return (SingleSource<T>) single;
     }
 
-    private static final class PublisherToPublisherSource<T> implements PublisherSource<T> {
+    // Visible for testing
+    static final class PublisherToPublisherSource<T> extends Publisher<T> implements PublisherSource<T> {
         private final Publisher<T> publisher;
 
         PublisherToPublisherSource(final Publisher<T> publisher) {
             this.publisher = requireNonNull(publisher);
         }
 
+        @Override
+        protected void handleSubscribe(final Subscriber<? super T> subscriber) {
+            deliverErrorFromSource(subscriber, newUnsupportedOperationException(getClass()));
+        }
+
+        @Override
+        void handleSubscribe(final Subscriber<? super T> subscriber,
+                             final CapturedContext capturedContext, final AsyncContextProvider contextProvider) {
+            publisher.handleSubscribe(subscriber, capturedContext, contextProvider);
```

Comment:
@bryce-anderson please carefully double-check that I implemented the right methods and didn't mess up anything by adding `extends`.